src/swap/letsexchange.js

```diff
@@ -0,0 +1,294 @@
+import {
+  type EdgeCorePluginOptions,
+  type EdgeCurrencyWallet,
+  type EdgeSpendInfo,
+  type EdgeSwapInfo,
+  type EdgeSwapPlugin,
+  type EdgeSwapQuote,
+  type EdgeSwapRequest,
+  type EdgeTransaction,
+  SwapBelowLimitError,
+  SwapCurrencyError
+} from 'edge-core-js/types'
+
+import {
+  type InvalidCurrencyCodes,
+  checkInvalidCodes,
+  makeSwapPluginQuote, getCodes
+} from '../swap-helpers.js'
+
+const pluginId = 'letsexchange'
+const swapInfo: EdgeSwapInfo = {
+  pluginId,
+  displayName: 'LetsExchange',
+  supportEmail: 'support@letsexchange.io'
+}
+
+const orderUri = 'https://api.letsexchange.io/?exchangeId='
+const uri = 'https://api.letsexchange.io/api/v1/'
+
+const expirationMs = 1000 * 60 * 20
+
+type QuoteInfo = {
+  transaction_id: string,
+  status: string,
+  coin_from: string,
+  coin_to: string,
+  network_from: string,
+  network_to: string,
+  deposit_amount: string,
+  withdrawal_amount: string,
+  deposit: string,
+  deposit_extra_id: string,
+  withdrawal: string,
+  withdrawal_extra_id: string,
+  rate: string,
+  fee: string,
+  return: string,
+  return_extra_id: string,
+  final_amount: string,
+  hash_in: string,
+  hash_out: string,
+  isEstimate: boolean
+}
+
+const dontUseLegacy = {
+  DGB: true
+}
+
+const INVALID_CURRENCY_CODES: InvalidCurrencyCodes = {
+  from: {
+    ETH: ['MATIC'],
+    AVAX: 'allTokens',
+    FTM: 'allTokens',
+    MATIC: 'allCodes'
+  },
+  to: {
+    ETH: ['MATIC'],
+    AVAX: 'allTokens',
+    FTM: 'allTokens',
+    MATIC: 'allCodes',
+    ZEC: ['ZEC']
+  }
+}
+
+async function getAddress(wallet: EdgeCurrencyWallet, currencyCode: string) {
+  const addressInfo = await wallet.getReceiveAddress({ currencyCode })
+  return addressInfo.legacyAddress && !dontUseLegacy[currencyCode]
+    ? addressInfo.legacyAddress
+    : addressInfo.publicAddress
+}
+
+function networkCodeConverter(networkCode: string): string {
+  switch (networkCode) {
+    case 'ETH':
+      networkCode = 'ERC20';
+      break;
+    case 'BNB':
+      networkCode = 'BEP2';
+      break;
+    case 'TRX':
+      networkCode = 'TRC20';
+      break;
+    case 'BSC':
+      networkCode = 'BEP20';
+      break;
+    default:
+      break;
+  }
+
+  return networkCode;
+}
+
+export function makeLetsExchangePlugin(
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const { initOptions, io, log } = opts
+  const { fetchCors = io.fetch } = io
+
+  async function call(url, request, data) {
+    const body = JSON.stringify(data.params)
+
+    const headers = {
+      'Content-Type': 'application/json',
+      Accept: 'application/json'
+    }
+    const response = await fetchCors(url, { method: 'POST', body, headers })
+    if (!response.ok) {
+      if (response.status === 422) {
+        throw new SwapCurrencyError(
+          swapInfo,
+          request.fromCurrencyCode,
+          request.toCurrencyCode
+        )
+      }
+      throw new Error(`letsexchange returned error code ${response.status}`)
```

Comment:
Are there any cases where the user might get an "over limit" error or "region not supported" error? If so, need to catch these and throw the appropriate error.

Reply:
No. There are not.